src/Compilers/Core/Portable/PEWriter/ManagedResource.cs

```diff
@@ -48,7 +48,11 @@ public void WriteData(BlobBuilder resourceWriter)
                         var count = (int)(stream.Length - stream.Position);
                         resourceWriter.WriteInt32(count);
 
-                        resourceWriter.WriteBytes(stream, count);
+                        int bytesWritten = resourceWriter.TryWriteBytes(stream, count);
+                        if (bytesWritten != count)
+                        {
+                            throw new EndOfStreamException($"Resource stream ended at {bytesWritten} bytes, expected {count} bytes");
```

Comment:
I was under the impression exception messages no longer had to be localizable. I also don't think we need to identify the stream since this message will be caught and turned into a ResourceException, which contains the name of the resource that failed to load.